diff --git a/src/com/cghs/stresstest/test/RebootTest.java b/src/com/cghs/stresstest/test/RebootTest.java
index b2a725c..2802841 100755
--- a/src/com/cghs/stresstest/test/RebootTest.java
+++ b/src/com/cghs/stresstest/test/RebootTest.java
@@ -116,7 +116,7 @@ public class RebootTest extends Activity implements OnClickListener {
 				updateBtnState();
 				mCountTV.setText(mCountTV.getText()+" TEST FINISH!");
 				
-			}else if(isRebootError()||isSystemError()){
+			}else if(isRebootError()){
 					mState = REBOOT_OFF;
 					saveSharedPreferences(mState, 0);
 					saveMaxTimes(0);
@@ -228,10 +228,19 @@ public class RebootTest extends Activity implements OnClickListener {
 					mCountDownTime--;
 					sendEmptyMessageDelayed(MSG_REBOOT_COUNTDOWN, 1000);
 				} else {
+				if(isSystemError()){
+					mState = REBOOT_OFF;
+					saveSharedPreferences(mState, 0);
+				    saveMaxTimes(0);
+					updateBtnState();
+				    mCountTV.setText(mCountTV.getText()+" Test fail for error!");
+							 
+				}else{
 					mCountdownTV.setText(getString(R.string.reboot_countdown)
 							+ mCountDownTime);
 					mCountdownTV.setVisibility(View.VISIBLE);
 					sendEmptyMessage(MSG_REBOOT);
+				   }
 				}
 
 				break;
diff --git a/src/com/cghs/stresstest/test/RecoveryTest.java b/src/com/cghs/stresstest/test/RecoveryTest.java
index 32bae92..e1f5c2f 100755
--- a/src/com/cghs/stresstest/test/RecoveryTest.java
+++ b/src/com/cghs/stresstest/test/RecoveryTest.java
@@ -55,6 +55,7 @@ public class RecoveryTest extends StressBase{
 	private boolean mIsEraseFlash = false;
 	private boolean mIsWipeAll = false;
 	
+	private boolean Ftest = false;
 	public static final int MSG_START_TEST = 1;
 	
 	@Override
@@ -77,7 +78,7 @@ public class RecoveryTest extends StressBase{
 		
 		if (mStartTest == 1) {
 			updateTestTimeTV();
-			if(isRebootError()||isSystemError()){
+			if(isRebootError()){
 					stopTest();
 					mTestTimeTv.setText(mTestTimeTv.getText()+" Test fail for error!");
 					 
@@ -118,6 +119,7 @@ public class RecoveryTest extends StressBase{
 	
 	@Override
 	public void onStartClick() {
+		Ftest=true;
 		preStartTest();
 	}
 
@@ -149,7 +151,12 @@ public class RecoveryTest extends StressBase{
 			@Override
 			public void onFinish() {
 				mCountdownTv.setVisibility(View.INVISIBLE);
+         if(!Ftest && isSystemError()){
+		  stopTest();
+		  mTestTimeTv.setText(mTestTimeTv.getText()+" Test fail for error!");
+		  }else{
 				startTest();
+				}
 			}
 		}.start();
 	}
